### DISCUSSION TITLE
NavMesh generation for gravity less agents

### DISCUSSION THREAD

**Discussion by skaiware** [recastnavigation/recastnavigation#565] [Q&A] [answered]
Hi
The nav mesh generator as today only handles agents under gravity capable to walk in respect to a max slope. 
https://github.com/recastnavigation/recastnavigation/blob/c5cbd53024c8a9d8d097a4371215e3342d2fdc87/Recast/Include/Recast.h#L224
Then
https://github.com/recastnavigation/recastnavigation/blob/c5cbd53024c8a9d8d097a4371215e3342d2fdc87/Recast/Source/Recast.cpp#L334
seems to mark the 'walkable' triangles given this maxSlope.

The swim navigation mode is just allowing the agent to fly, not having to be next to any collision triangle. 
There is though still a need to navigate an agent not linked to any gravity (infinite max slope, no up vector, ...) but with the  constraint to navigate next to a collision mesh (ground, wall, roof, whatever). An example is the spider style navigation. 
Is there a way to generate such nav mesh with recast ? 
Kind

Refs:
https://forums.unrealengine.com/t/navmesh-generation-past-90-degrees-or-wall-navmesh/11019/4

**Comment by grahamboree** [comment, marked as answer]
Not yet, unfortunately.  As you point out, Recast currently only supports "normal" ground-based navmeshes for humanoid agents that are affected by gravity and use the Y-axis as their up vector.  3d volumetric navigation for swimming/flying is something we're hoping to add to recast, since it can be extracted from the voxelization process.  We're also exploring ways to generate navigation data for climbing on surfaces, walls and ceilings.  Both are fairly involved features so they'd be a big effort and part of a larger release, rather than a quick addition for a point update.
